### PR TITLE
2016 Toyota Avalon

### PR DESCRIPTION
**Pull request by njbrown09**
Fully works without any errors.
2016 Toyota Avalon Limited with TSS-P
Toyota Giraffe on 0000 with Comma Power.
I was not able to test it without a DSU.

### CODE REVIEW

File: selfdrive/car/toyota/values.py [hunk 1]
```diff
@@ -141,6 +142,9 @@ def check_ecu_msgs(fingerprint, ecu):
...
   }],
+  CAR.AVALON: [{
+    36: 8, 37: 8, 170: 8, 180: 8, 426: 6, 452: 8, 464: 8, 466: 8, 467: 8, 547: 8, 550: 8, 552: 4, 562: 6, 608: 8, 610: 5, 643: 7, 705: 8, 800: 8, 835: 8, 836: 8, 849: 4, 869: 7, 870: 7, 896: 8, 897: 8, 905: 8, 911: 1, 916: 2, 921: 8, 933: 6, 944: 8, 945: 8, 951: 8, 955: 8, 956: 8, 979: 2, 1005: 2, 1014: 8, 1017: 8, 1041: 8, 1056: 8, 1059: 1, 1114: 8, 1176: 8, 1177: 8, 1178: 8, 1179: 8, 1180: 8, 1181: 8, 1190: 8, 1191: 8, 1192: 8, 1196: 8, 1200: 8, 1201: 8, 1202: 8, 1203: 8, 1206: 8, 1227: 8, 1237: 8, 1264: 8, 1279: 8, 1552: 8, 1553: 8, 1554: 8, 1555: 8, 1556: 8, 1557: 8, 1558: 8, 1561: 8, 1562: 8, 1568: 8, 1569: 8, 1570: 8, 1571: 8, 1572: 8, 1584: 8, 1589: 8, 1592: 8, 1593: 8, 1595: 8, 1596: 8, 1597: 8, 1664: 8, 1728: 8, 1779: 8, 1904: 8, 1912: 8, 1990: 8, 1998: 8
```
Comment (rbiasini) [inline selfdrive/car/toyota/values.py]: looks like this fingerprint was taken with camera disconnected (msg 740 is absent). It should be taken with stock system fully functional (giraffe setting to 1011).
See https://github.com/commaai/neo/tree/master/giraffe/toyota

Comment (njbrown09) [reply selfdrive/car/toyota/values.py]: Let me go do that.

Comment (njbrown09) [reply selfdrive/car/toyota/values.py]: Done

Comment (njbrown09) [reply selfdrive/car/toyota/values.py]: Hmm. It appears that having the camera in the fingerprint breaks it when I’m running 0000 on the giraffe. Would having two fingerprints be alright? One with stock (1001) and one with openpilot (0000)

**Comment by rbiasini** [comment]
Looks great. I could verify the correctness of the messages in `values.py`
Why you were not able to test it without DSU?

**Comment by njbrown09** [comment]
I was not able to test it without the DSU because the DSU is hidden and you are required to pull out the entire dashboard to get to it. If I can figure out how to pull out just the glove box and get it out I will but that won’t be for a fair amount of time.

File: selfdrive/car/toyota/values.py (hunk 1 above)
Comment (rbiasini) [reply selfdrive/car/toyota/values.py]: mmm... this is strange. the fingerprint you just added should be a super-set of the existing one.
I don't see a reason why it should break the behavior when giraffe is in 0000. Are you using the right switch config?
https://github.com/commaai/neo/tree/master/giraffe/toyota

Comment (njbrown09) [reply selfdrive/car/toyota/values.py]: I did the fingerprint for longer and it seems to be working now.

**Comment by njbrown09** [comment]
Vehicle is tuned and ready to be merged.

**Comment by njbrown09** [comment]
Here is a cabana drive.
https://community.comma.ai/cabana/?route=f7b6be73e3dfd36c%7C2019-05-14--15-18-53&max=16&url=https%3A%2F%2Fchffrprivate.blob.core.windows.net%2Fchffrprivate3%2Fv2%2Ff7b6be73e3dfd36c%2F65f6b506107c4a87cd5b4016c1e5b618_2019-05-14--15-18-53

Use the 2017 Avalon DBC

File: selfdrive/car/toyota/interface.py [hunk 2]
```diff
@@ -147,6 +147,16 @@ def get_params(candidate, fingerprint):
...
       ret.steerKf = 0.00006
+      
+    elif candidate == CAR.AVALON:
+      stop_and_go = False
+      ret.safetyParam = 100 # see conversion factor for STEER_TORQUE_EPS in dbc file
+      ret.wheelbase = 2.82
+      ret.steerRatio = 14.8 #Found at https://pressroom.toyota.com/releases/2016+avalon+product+specs.download
+      tire_stiffness_factor = 0.7983
+      ret.mass = 3505 * CV.LB_TO_KG + std_cargo  # mean between normal and hybrid
+      ret.steerKpV, ret.steerKiV = [[0.17], [0.03]]
```
Comment (rbiasini) [inline selfdrive/car/toyota/interface.py]: seems quite low. is it really better than standard 0.6/0.1?

Comment (njbrown09) [reply selfdrive/car/toyota/interface.py]: That’s what I though. It’s significantly better than 0.6/0.1

**Comment by njbrown09** [comment]
This is ready to go. Tuned it to the best i could!

**Comment by rbiasini** [comment]
yep, merged it internally. It will be in the next release. Thanks @njbrown09

**Pull request closed without merging**